aws-rotate-keys/orb.yml

```diff
@@ -43,4 +43,19 @@ commands:
             curl --fail https://circleci.com/api/v1.1/project/github/$CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME/envvar?circle-token=<< parameters.circleci-token >> \
               -X POST --header "Content-Type: application/json" \
               -d "{ \"name\":\"<< parameters.aws-secret-access-key-var >>\", \"value\":\"${new_secret_access_key}\" }" &&
-            aws iam delete-access-key --user-name << parameters.aws-username >> --access-key-id ${old_access_key_id}
+            aws iam delete-access-key --user-name << parameters.aws-username >> --access-key-id ${old_access_key_id} &&
+            echo &&
+            # Although the new token is visible on the aws API, aws does not accept it
+            # for authentication immediately.
+            AWS_ACCESS_KEY_ID=$new_access_key_id &&
```

Comment:
since we already make a post to update the aws variables (lines 40 and 43) do we still need to set these here?

Reply:
You need to set those to make aws cli use the new access key.

Reply:
As soon as we started using the rotate command we started getting InfoSec alerts about the new keys not yet used, so this is something all users of this command face. I think it worth solving it in a central place.
Another alternative would be to duplicate this logic in all users of this orb, and have it executed immediately after this rotate command, which is
- against DRY principle,
- hard to implement, since it is difficult to access the new value of the environment variables immediately after the rotate command execution (I couldn't find a way to force CircleCI to refresh the environment variables, although it might be possible by querying the CircleCI API with curl).
Do you have any alternative solution?

Reply:
sorry probably my question was not clear.
i am saying that in lines 40 and 43  we already set AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY (or we actually set the vars  the user set) so why do we need to do here `AWS_ACCESS_KEY_ID=$new_access_key_id`?
arent the values for the aws already updated?

i am think it is good to add the call here